dispatch/dispatch.go

```diff
@@ -0,0 +1,60 @@
+package dispatch
+
+import (
+	. "fmt"
+	"os"
+
+	"polydawn.net/repeatr/def"
+	"polydawn.net/repeatr/executor"
+	"polydawn.net/repeatr/executor/null"
+	"polydawn.net/repeatr/input"
+	"polydawn.net/repeatr/input/dir"
+	"polydawn.net/repeatr/input/tar"
+	"polydawn.net/repeatr/output"
+)
+
+// TODO: This should not require a global string -> class map :|
+// Should attempt to reflect-find, trying main package name first.
+// Will make simpler to use extended transports, etc.
```

Comment:
/shrug.  I'm fine with maintaining this as a string switch somewhere.  It's the path of least resistance and doesn't box anything in.  We might want characters that don't fit in a package name later; we might end up with some versioning scheme in the name that requires parsing; etc.  I've been following a rough convention of having a `$pkg.Type string` in every input package and moving on.  Another alternative is having a big globally visible map and having implementations register themselves in the package init function, but then you still have to *touch* the package, which means your import list just turned into a bunch of "_" imports and it's a bunch of strings you have to maintain again anyway.